projects/rocprofiler-sdk/source/include/rocprofiler-sdk/experimental/spm.h

```diff
@@ -0,0 +1,254 @@
+// MIT License
+//
+// Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+#include <rocprofiler-sdk/agent.h>
+#include <rocprofiler-sdk/defines.h>
+#include <rocprofiler-sdk/fwd.h>
+
+ROCPROFILER_EXTERN_C_INIT
+
+/**
+ * @brief (experimental) SPM parameter type used to configure SPM service.
+ *
+ **/
+typedef enum ROCPROFILER_SDK_EXPERIMENTAL rocprofiler_spm_parameter_type_t
+{
+    ROCPROFILER_SPM_PARAMETER_TYPE_SCLK_COUNT = 0,  ///< No. of sc clock counts to do sampling
+    ROCPROFILER_SPM_PARAMETER_TYPE_BUFFER_SIZE,     ///< SPM Buffer size, in bytes.
+    ROCPROFILER_SPM_PARAMETER_TYPE_TIMEOUT_MS,
+    ROCPROFILER_SPM_PARAMETER_TYPE_LAST
+    ///< SPM timeout in ms. Time to wait to read the SPM buffer
+} rocprofiler_spm_parameter_type_t;
+
+/**
+ * @brief SPM Profile Configurations
+ * @see rocprofiler_spm_create_counter_config for how to create.
+ */
+typedef struct ROCPROFILER_SDK_EXPERIMENTAL rocprofiler_spm_counter_config_id_t
+{
+    uint64_t handle;  ///< Opaque handle
+} rocprofiler_spm_counter_config_id_t;
+
+/**
+ * @brief (experimental) SPM parameter type and value.
+ *
+ **/
+typedef struct ROCPROFILER_SDK_EXPERIMENTAL rocprofiler_spm_parameter_t
+{
+    rocprofiler_spm_parameter_type_t type;   ///< SPM Parameter type
+    uint64_t                         value;  ///< SPM Parameter value
+} rocprofiler_spm_parameter_t;
```

Comment:
```suggestion
typedef struct ROCPROFILER_SDK_EXPERIMENTAL rocprofiler_spm_parameter_t
{
    uint64_t                         size;
    rocprofiler_spm_parameter_type_t type;   ///< SPM Parameter type
    uint64_t                         value;  ///< SPM Parameter value
} rocprofiler_spm_parameter_t;
```

If I am not mistaken... we may introduce new parameters in the future, which necessitates a `size` field. If we are certain we will never be adding to this struct, please explain why we will never need more than a enum and a uint64_t value.